com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest

PR run `master`

Failed in https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/34/testReport/junit/com.hazelcast.mapstore.postgres/PostgresGenericMapStoreTest/whenMapStoreDestroy_thenDropMapping/ 

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expecting:
  <[Row{[__map-store.people_o_72ec2bc5_8b57_4840_b37d_6fe0e0710b00]},
    Row{[__map-store.people_o_8ba88483_fbff_4bcb_940d_2b4ee583340e]}]>
to contain exactly in any order:
  <[Row{[__map-store.people_o_8ba88483_fbff_4bcb_940d_2b4ee583340e]}]>
but the following elements were unexpected:
  <[Row{[__map-store.people_o_72ec2bc5_8b57_4840_b37d_6fe0e0710b00]}]>

	at com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder(SqlTestSupport.java:322)
	at com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder(SqlTestSupport.java:284)
	at com.hazelcast.mapstore.GenericMapStoreTest.lambda$awaitMappingCreated$6(GenericMapStoreTest.java:627)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1236)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1253)
	at com.hazelcast.mapstore.GenericMapStoreTest.awaitMappingCreated(GenericMapStoreTest.java:626)
	at com.hazelcast.mapstore.GenericMapStoreTest.whenMapStoreDestroy_thenDropMapping(GenericMapStoreTest.java:113)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
:39:42,188 ERROR || - [InitExecutionOperation] hz.thirsty_lumiere.cached.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:333) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:552) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:333) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
15:39:42,199 ERROR || - [SubmitJobOperation] hz.thirsty_lumiere.cached.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:333) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:552) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:333) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
15:39:42,245 DEBUG || - [GenericMapStore] hz.dreamy_lumiere.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initializing for map people_o_8ba88483_fbff_4bcb_940d_2b4ee583340e
15:39:42,348 ERROR || - [InitExecutionOperation] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:333) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:552) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:333) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
15:39:42,349 ERROR || - [InitExecutionOperation] hz.thirsty_lumiere.cached.thread-4 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] CANCEL_FORCEFUL
com.hazelcast.jet.impl.exception.JobTerminateRequestedException: CANCEL_FORCEFUL
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:318) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:718) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution(JobExecutionService.java:714) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.operation.TerminateExecutionOperation.doRun(TerminateExecutionOperation.java:59) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:54) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:281) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:256) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:479) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:197) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:137) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.executeRun(OperationThread.java:123) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
15:39:42,351 ERROR || - [SubmitJobOperation] hz.dreamy_lumiere.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] null
java.util.concurrent.CancellationException: null
	at com.hazelcast.jet.sql.impl.processors.RootResultConsumerSink.tryProcess(RootResultConsumerSink.java:97) ~[hazelcast-sql-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:333) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:552) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:333) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[hazelcast-5.2-SNAPSHOT.jar:5.2-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
15:39:47,491  INFO |whenMapStoreDestroy_thenDropMapping| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
15:39:47,491  INFO |whenMapStoreDestroy_thenDropMapping| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
15:39:47,492  INFO |whenMapStoreDestroy_thenDropMapping| - [SimpleTestInClusterSupport] Time-limited test - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
15:39:47,492  INFO |whenMapStoreDestroy_thenDropMapping| - [SimpleTestInClusterSupport] Time-limited test - Destroying 2 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/__sql.catalog, hz:impl:flakeIdGeneratorService/__jet.ids]
BuildInfo right after whenMapStoreDestroy_thenDropMapping(com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest): BuildInfo{version='5.2-SNAPSHOT', build='20220823', buildNumber=20220823, revision=10a7ca9, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'whenMapStoreDestroy_thenDropMapping(com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest):'
15:39:40, accumulated pauses: 130 ms, max pause: 13 ms, pauses over 1000 ms: 0
15:39:45, accumulated pauses: 101 ms, max pause: 52 ms, pauses over 1000 ms: 0
```

</details>

--- a/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
+++ b/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
@@ -291,7 +291,7 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
 
     @Override
     public void destroy() {
-        awaitSuccessfulInit();
+        awaitInitFinished();
         dropMapping(mapping);
     }
 

--- a/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
+++ b/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
@@ -196,7 +196,7 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
             queries = new Queries(mapping, properties.idColumn, columnMetadataList);
         } catch (Exception e) {
             // We create the mapping on the first member initializing the MapStore
-            // Other members trying to concurrently initialize
+            // Other members trying to concurrently initialize will fail and just read the mapping
             if (e.getMessage() != null && e.getMessage().startsWith("Mapping or view already exists:")) {
                 readExistingMapping();
             } else {

--- a/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
+++ b/extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java
@@ -156,8 +156,6 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
 
         // Init can run on partition thread, creating a mapping uses other maps, so it needs to run elsewhere
         asyncExecutor.submit(() -> {
-            // We create the mapping only on the master node
-            // On other members we wait until the mapping has been created
             createMappingForMapStore(mapName);
         });
     }
@@ -197,6 +195,8 @@ public class GenericMapStore<K> implements MapStore<K, GenericRecord>, MapLoader
             }
             queries = new Queries(mapping, properties.idColumn, columnMetadataList);
         } catch (Exception e) {
+            // We create the mapping on the first member initializing the MapStore
+            // Other members trying to concurrently initialize
             if (e.getMessage() != null && e.getMessage().startsWith("Mapping or view already exists:")) {
                 readExistingMapping();
             } else {

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
@@ -25,6 +25,8 @@ import com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport;
 import com.hazelcast.jet.test.SerialTest;
 import com.hazelcast.nio.serialization.genericrecord.GenericRecord;
 import com.hazelcast.nio.serialization.genericrecord.GenericRecordBuilder;
+import com.hazelcast.sql.SqlResult;
+import com.hazelcast.sql.SqlRow;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.test.jdbc.H2DatabaseProvider;
@@ -41,6 +43,7 @@ import java.sql.DriverManager;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
@@ -61,8 +64,8 @@ import static org.assertj.core.util.Lists.newArrayList;
  * This test runs the MapStore methods directly, but it runs within real Hazelcast instance
  */
 @RunWith(HazelcastSerialClassRunner.class)
-@Category({ QuickTest.class, SerialTest.class })
-    public class GenericMapStoreTest extends JdbcSqlTestSupport {
+@Category({QuickTest.class, SerialTest.class})
+public class GenericMapStoreTest extends JdbcSqlTestSupport {
 
     public String mapName;
 
@@ -86,6 +89,14 @@ import static org.assertj.core.util.Lists.newArrayList;
             mapStore.destroy();
             mapStore = null;
         }
+
+        try (SqlResult result = sqlService.execute("SHOW MAPPINGS")) {
+            for (SqlRow next : result) {
+                Object name = next.getObject(0);
+                logger.warning("Mapping " + name + " exists after test, dropping");
+                sqlService.execute("DROP MAPPING IF EXISTS " + name).close();
+            }
+        }
     }
 
     @Test
@@ -93,7 +104,7 @@ import static org.assertj.core.util.Lists.newArrayList;
         createTable(mapName);
 
         mapStore = createMapStore();
-        awaitMappingCreated();
+        assertMappingCreated();
     }
 
     @Test
@@ -122,23 +133,23 @@ import static org.assertj.core.util.Lists.newArrayList;
         createTable(mapName);
 
         mapStore = createMapStore();
-        awaitMappingCreated();
+        assertMappingCreated();
 
         mapStore.destroy();
-        awaitMappingDestroyed();
+        assertMappingDestroyed();
     }
 
     @Test
-        public void whenMapStoreDestroyOnNonMaster_thenDropMapping() throws Exception {
+    public void whenMapStoreDestroyOnNonMaster_thenDropMapping() throws Exception {
         createTable(mapName);
 
         mapStore = createMapStore();
-        awaitMappingCreated();
+        assertMappingCreated();
 
         GenericMapStore<Object> mapStoreNotMaster = createMapStore(instances()[1]);
         mapStoreNotMaster.awaitInitFinished();
         mapStoreNotMaster.destroy();
-        awaitMappingDestroyed();
+        assertMappingDestroyed();
     }
 
     @Test
@@ -232,7 +243,7 @@ import static org.assertj.core.util.Lists.newArrayList;
         createTable(mapName, "id INT PRIMARY KEY", "name VARCHAR(100)", "age INT");
         executeJdbc("INSERT INTO " + mapName + " VALUES(0, 'name-0', 42)");
         createMapping(mapName, MAPPING_PREFIX + mapName);
-        awaitMappingCreated();
+        assertMappingCreated();
         // This simulates a second map store on a different instance. The mapping is created, but must be validated
         // (e.g. the config might differ on members)
         Properties secondProps = new Properties();
@@ -245,7 +256,6 @@ import static org.assertj.core.util.Lists.newArrayList;
         assertThatThrownBy(() -> mapStore.load(0))
                 .isInstanceOf(HazelcastException.class)
                 .hasStackTraceContaining("Column 'age' not found");
-        mapStore.destroy();
     }
 
     @Test
@@ -647,6 +657,10 @@ import static org.assertj.core.util.Lists.newArrayList;
         return createMapStore(properties, instance);
     }
 
+    private <K> GenericMapStore<K> createMapStore(Properties properties, HazelcastInstance instance) {
+        return createMapStore(properties, instance, true);
+    }
+
     private <K> GenericMapStore<K> createMapStore(Properties properties, HazelcastInstance instance, boolean init) {
         MapConfig mapConfig = createMapConfigWithMapStore(mapName);
         instance.getConfig().addMapConfig(mapConfig);
@@ -659,10 +673,6 @@ import static org.assertj.core.util.Lists.newArrayList;
         return mapStore;
     }
 
-    private <K> GenericMapStore<K> createMapStore(Properties properties, HazelcastInstance instance) {
-        return createMapStore(properties, instance, true);
-    }
-
     private static MapConfig createMapConfigWithMapStore(String mapName) {
         MapStoreConfig mapStoreConfig = new MapStoreConfig();
         mapStoreConfig.setClassName(GenericMapStore.class.getName());
@@ -671,15 +681,13 @@ import static org.assertj.core.util.Lists.newArrayList;
         return mapConfig;
     }
 
-    private void awaitMappingCreated() {
+    private void assertMappingCreated() {
         assertTrueEventually(() -> {
             assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList(new Row(MAPPING_PREFIX + mapName)));
         }, 60);
     }
 
-    private void awaitMappingDestroyed() {
-        assertTrueEventually(() -> {
-            assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList());
-        }, 60);
+    private void assertMappingDestroyed() {
+        assertTrueEventually(() -> assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList()), 60);
     }
 }

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
@@ -25,8 +25,6 @@ import com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport;
 import com.hazelcast.jet.test.SerialTest;
 import com.hazelcast.nio.serialization.genericrecord.GenericRecord;
 import com.hazelcast.nio.serialization.genericrecord.GenericRecordBuilder;
-import com.hazelcast.sql.SqlResult;
-import com.hazelcast.sql.SqlRow;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.test.jdbc.H2DatabaseProvider;
@@ -88,14 +86,6 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
             mapStore.destroy();
             mapStore = null;
         }
-
-        try (SqlResult result = sqlService.execute("SHOW MAPPINGS")) {
-            for (SqlRow next : result) {
-                Object name = next.getObject(0);
-                logger.warning("Mapping " + name + " exists after test, dropping");
-                sqlService.execute("DROP MAPPING IF EXISTS " + name).close();
-            }
-        }
     }
 
     @Test
@@ -146,7 +136,6 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         assertMappingCreated();
 
         GenericMapStore<Object> mapStoreNotMaster = createMapStore(instances()[1]);
-        mapStoreNotMaster.awaitInitFinished();
         mapStoreNotMaster.destroy();
         assertMappingDestroyed();
     }

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
@@ -234,7 +234,6 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         assertThatThrownBy(() -> mapStore.load(0))
                 .isInstanceOf(HazelcastException.class)
                 .hasStackTraceContaining("Column 'age' not found");
-        mapStore = null;
     }
 
     @Test
@@ -250,7 +249,6 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         secondProps.setProperty(COLUMNS_PROPERTY, "id,name,age");
         mapStore = createMapStore(secondProps, hz, false);
         mapStore.init(hz, secondProps, mapName);
-        mapStore.awaitInitFinished();
 
         assertThatThrownBy(() -> mapStore.load(0))
                 .isInstanceOf(HazelcastException.class)

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
@@ -43,7 +43,6 @@ import java.sql.DriverManager;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.util.HashMap;
-import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;

--- a/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
+++ b/extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java
@@ -22,10 +22,10 @@ import com.hazelcast.core.HazelcastException;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.internal.serialization.impl.compact.CompactGenericRecord;
 import com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport;
+import com.hazelcast.jet.test.SerialTest;
 import com.hazelcast.nio.serialization.genericrecord.GenericRecord;
 import com.hazelcast.nio.serialization.genericrecord.GenericRecordBuilder;
 import com.hazelcast.test.HazelcastSerialClassRunner;
-import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.test.jdbc.H2DatabaseProvider;
 import org.junit.After;
@@ -61,8 +61,8 @@ import static org.assertj.core.util.Lists.newArrayList;
  * This test runs the MapStore methods directly, but it runs within real Hazelcast instance
  */
 @RunWith(HazelcastSerialClassRunner.class)
-@Category({QuickTest.class, ParallelJVMTest.class})
-public class GenericMapStoreTest extends JdbcSqlTestSupport {
+@Category({ QuickTest.class, SerialTest.class })
+    public class GenericMapStoreTest extends JdbcSqlTestSupport {
 
     public String mapName;
 
@@ -101,7 +101,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         createTable(mapName);
         insertItems(mapName, 1);
 
-        GenericMapStore<Object> mapStore = createMapStore(instances()[1]);
+        mapStore = createMapStore(instances()[1]);
         GenericRecord record = mapStore.load(0);
         assertThat(record).isNotNull();
     }
@@ -129,13 +129,14 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
     }
 
     @Test
-    public void whenMapStoreDestroyOnNonMaster_thenDropMapping() throws Exception {
+        public void whenMapStoreDestroyOnNonMaster_thenDropMapping() throws Exception {
         createTable(mapName);
 
         mapStore = createMapStore();
         awaitMappingCreated();
 
         GenericMapStore<Object> mapStoreNotMaster = createMapStore(instances()[1]);
+        mapStoreNotMaster.awaitInitFinished();
         mapStoreNotMaster.destroy();
         awaitMappingDestroyed();
     }
@@ -231,19 +232,20 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         createTable(mapName, "id INT PRIMARY KEY", "name VARCHAR(100)", "age INT");
         executeJdbc("INSERT INTO " + mapName + " VALUES(0, 'name-0', 42)");
         createMapping(mapName, MAPPING_PREFIX + mapName);
-
+        awaitMappingCreated();
         // This simulates a second map store on a different instance. The mapping is created, but must be validated
         // (e.g. the config might differ on members)
         Properties secondProps = new Properties();
         secondProps.setProperty(EXTERNAL_REF_ID_PROPERTY, TEST_DATABASE_REF);
         secondProps.setProperty(COLUMNS_PROPERTY, "id,name,age");
-        mapStore = createMapStore(secondProps, hz);
+        mapStore = createMapStore(secondProps, hz, false);
         mapStore.init(hz, secondProps, mapName);
+        mapStore.awaitInitFinished();
 
         assertThatThrownBy(() -> mapStore.load(0))
                 .isInstanceOf(HazelcastException.class)
                 .hasStackTraceContaining("Column 'age' not found");
-        mapStore = null;
+        mapStore.destroy();
     }
 
     @Test
@@ -347,7 +349,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
     @Test
     public void givenRowDoesNotExist_whenLoadAll_thenReturnEmptyMap() throws Exception {
         createTable(mapName);
-        GenericMapStore<Integer> mapStore = createMapStore();
+        mapStore = createMapStore();
 
         Map<Integer, GenericRecord> records = mapStore.loadAll(newArrayList(0));
         assertThat(records).isEmpty();
@@ -356,7 +358,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
     @Test
     public void givenRow_whenLoadAllKeys_thenReturnKeys() throws Exception {
         createTable(mapName);
-        GenericMapStore<Integer> mapStore = createMapStore();
+        mapStore = createMapStore();
 
         insertItems(mapName, 1);
 
@@ -382,7 +384,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
     @Test
     public void givenNoRows_whenLoadAllKeys_thenEmptyIterable() throws Exception {
         createTable(mapName);
-        GenericMapStore<Integer> mapStore = createMapStore();
+        mapStore = createMapStore();
 
         Iterable<Integer> ids = mapStore.loadAllKeys();
         assertThat(ids).isEmpty();
@@ -412,7 +414,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
     @Test
     public void givenIdColumn_whenStore_thenTableContainsRow() throws Exception {
         createTable(mapName);
-        GenericMapStore<Integer> mapStore = createMapStore();
+        mapStore = createMapStore();
 
         GenericRecord person = GenericRecordBuilder.compact("Person")
                                                    .setInt32("id", 0)
@@ -491,7 +493,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
     @Test
     public void whenStoreAllWithNoRecords_thenDoNothing() throws Exception {
         createTable(mapName);
-        GenericMapStore<Integer> mapStore = createMapStore();
+        mapStore = createMapStore();
 
         mapStore.storeAll(emptyMap());
 
@@ -520,7 +522,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         properties.setProperty(EXTERNAL_REF_ID_PROPERTY, TEST_DATABASE_REF);
 
         properties.setProperty(ID_COLUMN_PROPERTY, "person-id");
-        GenericMapStore<Integer> mapStore = createMapStore(properties, hz);
+        mapStore = createMapStore(properties, hz);
         mapStore.delete(0);
 
         assertJdbcRowsAnyOrder(mapName,
@@ -620,7 +622,7 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         properties.setProperty(EXTERNAL_REF_ID_PROPERTY, TEST_DATABASE_REF);
 
         properties.setProperty("columns", "id,name");
-        GenericMapStore<Integer> mapStore = createMapStore(properties, hz);
+        mapStore = createMapStore(properties, hz);
 
         GenericRecord person = GenericRecordBuilder.compact(mapName)
                 .setInt32("id", 1)
@@ -645,14 +647,20 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
         return createMapStore(properties, instance);
     }
 
-    private <K> GenericMapStore<K> createMapStore(Properties properties, HazelcastInstance instance) {
+    private <K> GenericMapStore<K> createMapStore(Properties properties, HazelcastInstance instance, boolean init) {
         MapConfig mapConfig = createMapConfigWithMapStore(mapName);
         instance.getConfig().addMapConfig(mapConfig);
 
         GenericMapStore<K> mapStore = new GenericMapStore<>();
-        mapStore.init(instance, properties, mapName);
-        mapStore.awaitInitFinished();
+        if (init) {
+            mapStore.init(instance, properties, mapName);
+            mapStore.awaitInitFinished();
+        }
         return mapStore;
+    }
+
+    private <K> GenericMapStore<K> createMapStore(Properties properties, HazelcastInstance instance) {
+        return createMapStore(properties, instance, true);
     }
 
     private static MapConfig createMapConfigWithMapStore(String mapName) {
@@ -666,12 +674,12 @@ public class GenericMapStoreTest extends JdbcSqlTestSupport {
     private void awaitMappingCreated() {
         assertTrueEventually(() -> {
             assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList(new Row(MAPPING_PREFIX + mapName)));
-        }, 5);
+        }, 60);
     }
 
     private void awaitMappingDestroyed() {
         assertTrueEventually(() -> {
             assertRowsAnyOrder(hz, "SHOW MAPPINGS", newArrayList());
-        }, 5);
+        }, 60);
     }
 }

